src/main/scala/com/fulcrumgenomics/commons/io/Io.scala

```diff
@@ -55,12 +56,16 @@ trait IoUtil {
 
   /** Creates a new InputStream to read from the supplied path. Automatically handles gzipped files. */
   def toInputStream(path: Path) : InputStream = {
+
     PathUtil.extensionOf(path) match {
       case Some(".gz") | Some(".bgz") | Some(".bgzip") =>
         new GZIPInputStream(Files.newInputStream(path), bufferSize)
       case _ => {
         val stream =  if (Files.isSameFile(path, Io.StdIn)) System.in else Files.newInputStream(path)
+        val attrs  = Files.readAttributes(path.toRealPath(), classOf[BasicFileAttributes])
         new BufferedInputStream(stream, bufferSize)
+        if (!attrs.isRegularFile && attrs.isOther) stream
```

Comment:
Can you add a comment here that this is to support named pipes?  And I have no recollection why we don't buffer named pipes, do you?